pkg/operator/controllers/imageconfig/image_controller.go

```diff
@@ -119,18 +119,20 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 // Switch case to ensure the correct registries are added depending on the cloud environment (Gov or Public cloud)
 func getCloudAwareRegistries(instance *arov1alpha1.Cluster) ([]string, error) {
-	var requiredRegistries []string
+	acrSubdomain := strings.Split(instance.Spec.ACRDomain, ".")[0]
```

Comment:
This is set (for prod) in [pkg/env/prod.go](https://github.com/Azure/ARO-RP/blob/fc656bc7a8feb251227a170186c9ddbb90c5c064/pkg/env/prod.go#L175-L185) by querying the Azure endpoint and building the domain name. This is a quick way of getting `acrResource.ResourceName` without having to re-query the Azure API.

Reply:
Perhaps a quick error check that the resultant string is of a minimum length? In future, a more robust check would be ensuring the string conforms to a DNS label spec.

Reply:
Addressed in changes. I also included a quick check for `instance.Spec.ACRDomain` that we were already returning as part of the required registries since that's where any problem would originate from.